### ISSUE TITLE
source blocks always numbered as figures - any way to achieve a different category?

### ISSUE THREAD

**Issue by jechterhoff** [metanorma/metanorma-ogc#461]
Short version: It looks like source blocks are always numbered as figures by metanorma. I would like to number them as listings.

Long version: As documented at https://www.metanorma.org/author/topics/document-format/text/#unnumbered-elements: _"Sourcecode and pseudocode snippets are by default numbered as figures"_. The only way to suppress that would be to use the `%unnumbered` option attribute. What I would like to do is to assign a new numbered category for the source blocks in my document, named “Listing”. With that, I would be able to assign a title to a source block (which is also possible using %unnumbered), but together with a listing number.

A test project (as well as further information about the metanorma version and command I used) is available in #460.

**Comment by ronaldtse** [comment]
@jechterhoff Thank you for the suggestion!

While we are always open to feedback, please understand that Metanorma is a standardized content model used by SDOs, is *tightly tied to requirements of the SDOs* and does not allow exceptions.

In the case of the OGC standards authoring format, OGC does not have a document element called "Listing". For authoring rules that are not defined at OGC, it adopts many authoring requirements from the ISO/IEC Directives Part 2. Here, this aligns with ISO practice where all source code blocks are "Figures". Metanorma does not allow custom document element categories.

In short, this question is about OGC standards policy.

This question of a new document element will:
1. Need to be approved by the OGC DocTeam as a matter of policy
2. Need to be implemented as an OGC-wide change. i.e. in all OGC standards, the "source code" block will always be a Listing, not a Figure.

Thanks!

**Comment by jechterhoff** [comment]
@ronaldtse Sure, I understand that the category would have to be "officially" recognized by SDOs. To me, it feels alien to mark a source code block as "Figure". 

@ghobona: What is the position of OGC in this matter?

**Comment by ghobona** [comment]
[OGC Editorial guidance](https://github.com/opengeospatial/templates/blob/master/editor_guidance.adoc) does not currently mention Listings, so I have created a [GitHub Issue](https://github.com/opengeospatial/templates/issues/37) in the relevant repository for the issue to be discussed and resolved.

**Comment by ronaldtse** [comment]
Thank you @ghobona !

**Comment by ogcscotts** [comment]
I can add this discussion to the DocTeam with the proposal that all source code NOT be figures. I am not sure about the context of a "listing" - are there any other elements that are tagged as "listings?"

**Comment by ghobona** [comment]
Historically, it's only source code and fragments of markup that were tagged as listings. 

Some examples are [CityGML 3.0 Part 2](https://docs.ogc.org/is/21-006r2/21-006r2.html#listing-gml-identifiers), [IMIS Recommendations ER](https://docs.ogc.org/per/15-118r1.html), and [Testbed-14 JSON ER](https://docs.ogc.org/per/18-091r2.html#JSON_Schema_Conversion_composition).

**Comment by opoudjis** [comment]
@ghobona @ogcscotts I am going away on holiday from 16 Jan until 29 Jan, and I expect we will catch up in early Feb as normal. A reminder that this issue is still pending.

**Comment by opoudjis** [comment]
Expect DocTeam resolution by end March

**Comment by ghobona** [comment]
DocTeam telecon likely in April.

**Comment by ghobona** [comment]
DocTeam telecon in April approved this proposal.

**Comment by ghobona** [comment]
The DocTeam agreed on a resolution regarding source blocks.

The resolution is...

- **if** `[%unnumbered%]` is used **OR** the source is inline, **then** do not use any label.
- **else** use the "Listing" label.

The associated comment is https://github.com/opengeospatial/templates/issues/37#issuecomment-2083148861

**Comment by ghobona** [comment]
Enhancement verified.
